src/Poseidon/EntitiesList.hs

```diff
@@ -78,6 +81,19 @@ instance EntitySpec PoseidonEntity where
         parseInd   = Ind   <$> P.between (P.char '<') (P.char '>') parseName
         parseName  = P.many1 (P.satisfy (\c -> not (isSpace c || c `elem` ",<>*")))
 
+-- turns out that we cannot easily write instances for classes, so need to be explicit for both types
+instance FromJSON PoseidonEntity where parseJSON = withText "PoseidonEntity" aesonParseEntitySpec
+instance FromJSON SignedEntity   where parseJSON = withText "SignedEntity" aesonParseEntitySpec
+instance ToJSON   PoseidonEntity where toJSON e = String (pack $ show e)
+instance ToJSON   SignedEntity   where toJSON e = String (pack $ show e)
+
+aesonParseEntitySpec :: (EntitySpec e) => Text -> Parser e
```

Comment:
What's the advantage of `Text` here over `String`, if you have to unpack anyway? I read that Text is better than String in many ways, but here it seems a bit sporadic?